### PR TITLE
[8.x] Add `insertBetweenEach()` collection method

### PR DESCRIPTION
**Pull request by jeroenvanrensen**
Hello,

I added the `insertBetweenEach()` collection method.

This method allows you to insert an item into a collection between each existing item.

For example:

```php 
collect(['a', 'b', 'c'])->insertBetweenEach('x'); // ['a', 'x', 'b', 'x', 'c']
```

This can replace this code:

```php
collect(['a', 'b', 'c'])->flatMap(function ($item) {
   return [$item, 'x'];   
})->slice(0, -1); // ['a', 'x', 'b', 'x', 'c']
```

I'm not really sure about the name of the method, but I couldn't think of anything better.

Thanks! Jeroen

### CODE REVIEW

File: src/Illuminate/Collections/Collection.php [hunk 1]
```diff
@@ -521,6 +521,19 @@ public function implode($value, $glue = null)
...
+     * Insert a value between each item in the collection.
+     * 
+     * For example: ['a', 'b', 'c'] with 'x' becomes ['a', 'x', 'b', 'x', 'c'].
+     *
+     * @param  string  $value
+     * @return static
+     */
+    public function insertBetweenEach($value)
+    {
+        return new static($this->flatMap(fn ($item) => [$item, $value])->slice(0, -1));
```
Comment (GrahamCampbell) [inline src/Illuminate/Collections/Collection.php:534]: this seems pretty niche. why not just wrote `->flatMap(fn ($item) => [$item, $value])->slice(0, -1)` in your app?

**Review by GrahamCampbell — CHANGES_REQUESTED** [review]

File: src/Illuminate/Collections/Collection.php (hunk 1 above)
Comment (jeroenvanrensen) [reply src/Illuminate/Collections/Collection.php:534]: @GrahamCampbell because it's not very clear what it's doing

Comment (GrahamCampbell) [reply src/Illuminate/Collections/Collection.php:534]: then register a macro in your app?

Comment (jeroenvanrensen) [reply src/Illuminate/Collections/Collection.php:534]: Yes that's possible but I think others can benefit from this as well

**Comment by taylorotwell** [comment]
Thanks for your pull request to Laravel!

Unfortunately, I'm going to delay merging this code for now. To preserve our ability to adequately maintain the framework, we need to be very careful regarding the amount of code we include. 

If possible, please consider releasing your code as a package so that the community can still take advantage of your contributions! 

If you feel absolutely certain that this code corrects a bug in the framework, please "**@**" mention me in a follow-up comment with further explanation so that GitHub will send me a notification of your response.

**Comment by jeroenvanrensen** [comment]
@taylorotwell can you please explain why you closed it?

**Pull request closed without merging**